src/nyc_trees/libs/auth_backend.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+from django.contrib.auth.backends import ModelBackend
+
+from apps.core.models import User
+
+
+class EmailOrUsernameAuthentication(ModelBackend):
+    def authenticate(self, username=None, password=None):
+        user = None
+        try:
```

Comment:
just to be safe - are we sure that `@` and `.` are not allowed in usernames?
I'm imagining a case with username/password/emails: `foo@bar.com / qux / treesRgood@gmail.com` and `lenny123 / qux / foo@bar.com`. The former user could accidentally log into the latter user's account.